src/data/openai_client.rs

```diff
@@ -136,60 +112,8 @@ impl OpenAIClient {
     ///
     /// The loop continues until the command channel is closed or an unrecoverable error occurs.
     fn process_agent_commands(command_receiver: mpsc::Receiver<ProviderCommand>, address: String, api_key: Option<String>) {
-        let rt = tokio::runtime::Runtime::new().unwrap();
-        let mut current_request: Option<JoinHandle<()>> = None;
-
         while let Ok(command) = command_receiver.recv() {
             match command {
-                ProviderCommand::SendMessage(task, model, tx) => {
```

Comment:
Since these clients are now only fetching models, their API (and the traits they're implementing, commands, etc) can be simplified a lot.

Ideally we would use MolyKit features to fetch for the models, there where some edge cases but I think now it might be replaceable. The main edge case (for why I didn't initially wire things like the providers dashboard with MolyKit `bots()`) is MoFa and DeepInquire not supporting /v1/models, so their clients hardcode the response, we should work around it and use MolyKit clients instead.
Since it's now a huge deal right now, I don't want to block this PR with this issue and kept these clients around, we can iterate over this once merged.